scripts/joint_trajectory_file_playback.py

```diff
@@ -228,7 +230,18 @@ def find_start_offset(pos):
             cmd, values = self._clean_line(values, joint_names)
             #find allowable time offset for move to start position
             if idx == 0:
+                # Set the initial position to be the current pose.
+                # This ensures we move slowly to the starting point of the
+                # trajectory from the current pose - The user may have moved
```

Comment:
The Joint Trajectory Action Server uses a position's `time_from_start` as the time when that position needs to be achieved, dictating the speed of a movement by the delta in position between `time_from_start`'s. Here we are hard coding the `time_from_start` for the starting current position to be `0.0`.

Then the `start_offset` time is calculated to figure out the amount of time required from the initial pose to the starting point:
`224    start_offset = find_start_offset(cmd)`
That aligns the  end of the required safe movement with the start of trajectory playback.